tldw_Server_API/app/core/DB_Management/chacha/conversation_store.py

```
@@ -0,0 +1,1322 @@
+    def _normalize_scope(
+        self,
+        scope_type: str | None,
+        workspace_id: str | None,
+    ) -> tuple[str, str | None]:
+        if scope_type is None:
+            scope_type = "global"
+        scope_type = scope_type.strip().lower()
+        if scope_type not in self._db._ALLOWED_SCOPE_TYPES:
+            raise InputError(
+                f"Invalid scope_type '{scope_type}'. Allowed: {', '.join(self._db._ALLOWED_SCOPE_TYPES)}"
+            )  # noqa: TRY003
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)